task1.c

```diff
@@ -0,0 +1,118 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+
+void swap(int *firstElement, int *secondElement)
+{
+    int transitElement = *firstElement;
+    *firstElement = *secondElement;
+    *secondElement = transitElement;
+}
+
+void printArray(int *array, int size)
+{
+    for (int i = 0; i < size; ++i)
+    {
+        printf("%d ", array[i]);
+    }
+    printf("\n");
+}
+
+int elementsPermutation(int *array, int arrayLength)
+{
+    int counterOfPermutations = 0;
+    int j = 0;
+    for (int i = 0; i < arrayLength; ++i)
+    {
+        if (array[i] % 2 == 0)
+        {
+            if (array[i - 1] % 2 == 0 || i == 0)
```

Comment:
```suggestion
            if (i == 0 || array[i - 1] % 2 == 0)
```
Порядок тут важен. Если первый дизъюнкт вычислится в `true`, то второй даже вычисляться не будет. Поэтому при `i = 0` не произайдет выхода за пределы массива.
А при исходной реализации будет выход за пределы массива.